cs-google/src/main/scala/io/cloudslang/content/google/services/compute/disks/DiskController.scala

```diff
@@ -0,0 +1,79 @@
+package io.cloudslang.content.google.services.compute.disks
+
+import com.google.api.services.compute.model._
+import io.cloudslang.content.utils.CollectionUtilities.toList
+
+/**
+  * Created by victor on 3/5/17.
+  */
+object DiskController {
+  def createDisk(zone: String, diskName: String, sourceImageOpt: Option[String], snapshotImageOpt: Option[String], imageEncryptionKeyOpt: Option[String],
+                 diskEncryptionKeyOpt: Option[String], diskType: String, diskDescription: String, licensesList: String,
+                 licensesDel: String, diskSize: Long): Disk = {
+    val computeDisk = new Disk()
+      .setName(diskName)
+      .setDescription(diskDescription)
+      .setZone(zone)
+      .setType(diskType)
+      .setLicenses(toList(licensesList, licensesDel))
+      .setSizeGb(diskSize)
+
+    if (sourceImageOpt.isDefined) {
+      computeDisk.setSourceImage(sourceImageOpt.get)
+      imageEncryptionKeyOpt.foreach { encrypt => computeDisk.setSourceImageEncryptionKey(new CustomerEncryptionKey().setRawKey(encrypt)) }
+    } else if (snapshotImageOpt.isDefined) {
+      computeDisk.setSourceSnapshot(snapshotImageOpt.get)
+      imageEncryptionKeyOpt.foreach { encrypt => computeDisk.setSourceSnapshotEncryptionKey(new CustomerEncryptionKey().setRawKey(encrypt)) }
+    }
+    diskEncryptionKeyOpt.foreach { encrypt => computeDisk.setDiskEncryptionKey(new CustomerEncryptionKey().setRawKey(encrypt)) }
+    computeDisk
+  }
+
+  def createAttachedDisk(boot:Boolean, autoDelete: Boolean, deviceNameOpt: Option[String], mode: String, source: String, interface: String): AttachedDisk = {
+    val attachedDisk = new AttachedDisk()
+      .setBoot(boot)
+      .setAutoDelete(autoDelete)
+      .setMode(mode)
+      .setSource(source)
+      .setInterface(interface)
+
+    deviceNameOpt match {
+      case Some(deviceName) => attachedDisk.setDeviceName(deviceName)
+      case _ => attachedDisk
+    }
+  }
+
+  def createAttachedDisk(boot: Boolean,
```

Comment:
method overloading ... you could have used a method with default values and pattern matching ( like the other methods)